otter/metrics.py

```diff
@@ -253,6 +254,40 @@ def connect_cass_servers(reactor, config):
         seed_endpoints, config['keyspace'], disconnect_on_cancel=True)
 
 
+def log_divergent_groups(clock, divergent_groups, log, timeout, group_metrics):
+    """
+    Log groups that have not changed and been divergent for long time
+    """
+    dg = divergent_groups
+    converged, diverged = partition_bool(
+        lambda gm: gm.actual + gm.pending == gm.desired, group_metrics)
+    # stop tracking all converged groups
+    for gm in converged:
+        dg.pop((gm.tenant_id, gm.group_id), None)
+    # Start tracking divergent groups depending on whether they've changed
+    now = clock.seconds()
+    for gm in diverged:
+        pair = (gm.tenant_id, gm.group_id)
+        if pair in dg:
+            last_time, values = dg[pair]
+            if values != hash((gm.desired, gm.actual, gm.pending)):
+                del dg[pair]
```

Comment:
Thanks for the detailed explanation and sorry you had to do that. I should've expressed it better earlier. The log says `Group has been diverged _and_ unchanged for x time` which is off by `m` seconds. Considering that `m` is small, I think its fine. Moreover, this information is for diagnostic purpose only. I would imagine us to look for groups logs in kibana and find out why it is divergent in which case exact time is not really an issue. To be clear, I completely agree with you about the incorrect time :) but I think its fine.